### PR TITLE
mgr/dashboard: List gateways in a group 

### PR DESCRIPTION
**Pull request by afreen23**
- adds a group selection component for gateways
- fetch gateway using daemons info
- added tests

fixes https://tracker.ceph.com/issues/68135

## Demo

[Screencast from 2024-09-19 15-35-09.webm](https://github.com/user-attachments/assets/44c5bcec-5927-4bbf-8c67-3c62a1bdf47c)

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by afreen23** [comment]
jenkins test api

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts [hunk 1]
```diff
@@ -2,43 +2,98 @@ import { Component } from '@angular/core';
...
-        name: $localize`Address`,
-        prop: 'addr'
+        name: $localize`Host name`,
+        prop: 'hostname'
       },
       {
-        name: $localize`Port`,
-        prop: 'port'
+        name: $localize`Status`,
+        prop: 'status'
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts:53]: do we have a status and hostname to show in the output of `/api/nvmeof/gateway`?

Comment (afreen23) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts:53]: This is just daemon status, similar to what one would see in the service list page.
We dont have status info from `/api/nvmeof/gateway` API as of now. Its just some metadata, like addr, port and gateway ID.

Moving forward we need to align the UI view and API with the CLI version which gives useful information about gateways:

```
           "gw-id": "client.nvmeof.nvmeof.default.ceph-node-01.kdcguk",
            "anagrp-id": 2,
            "num-namespaces": 0,
            "performed-full-startup": 1,
            "Availability": "AVAILABLE",
            "ana states": " 1: STANDBY ,  2: ACTIVE "
```

Comment (afreen23) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts:53]: FYI the gateway output from `/api/nvmeof/gateway`

```
{
  "cli_version": "",
  "version": "1.2.17",
  "name": "client.nvmeof.nvmeof.default.ceph-node-02.hybprc",
  "group": "default",
  "addr": "192.168.100.102",
  "port": "5500",
  "load_balancing_group": 1,
  "spdk_version": "24.01"
}

```

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
Update: Added more tests for gateways and fixed console errors in tests.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts:53]: alright, thanks

Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway/nvmeof-gateway.component.ts:53]: shall we have the badge here like we have in the daemon list?

**Review by nizamial09 — APPROVED** [review]
thanks again @afreen23

**Comment by afreen23** [comment]
Update: added badge for daemon status, check the updated demo.

**Comment by nizamial09** [comment]
> Update: added badge for daemon status, check the updated demo.

thank you. lgtm

**Comment by afreen23** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by rivkap** [comment]
@afreen23 @nizamial09 note that "host name" should be "hostname", if possible per ibm style: https://ibmdocs-test.dcs.ibm.com/docs/en/ibm-style?topic=word-usage#host-name

**Pull request merged**